compiler/env/RegionProfiler.hpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+   *
+   * (c) Copyright IBM Corp. 2017, 2017
+   *
+   *  This program and the accompanying materials are made available
+   *  under the terms of the Eclipse Public License v1.0 and
+   *  Apache License v2.0 which accompanies this distribution.
+   *
+   *      The Eclipse Public License is available at
+   *      http://www.eclipse.org/legal/epl-v10.html
+   *
+   *      The Apache License v2.0 is available at
+   *      http://www.opensource.org/licenses/apache2.0.php
+   *
+   * Contributors:
+   *    Multiple authors (IBM Corp.) - initial implementation and documentation
+   ******************************************************************************/
+
+#ifndef OMR_REGION_PROFILER_HPP
+#define OMR_REGION_PROFILER_HPP
+
+#pragma once
+
+#include "env/Region.hpp"
+#include "env/SegmentProvider.hpp"
+#include "compile/Compilation.hpp"
+
+namespace TR {
+
```

Comment:
I think some level of Doxygen commentary is appropriate here.  Unstated, but implied by this feature as well is that its only appropriate if you have a `TR::Compilation` in hand.